### PR TITLE
[ 3° ] - Release/v1.4.0 Bugfix/dtr lookup: Fixed bugs related to negotiation timeout and "gost" contract id

### PR DESCRIPTION
**Pull request by matbmoser**
# Why we create this PR?
 
There is a bug related to the search of the digital twin registries, sometimes there is coming no contract id and the backend enters in a loop without retrying the negotiation of contract.
 
# What we want to achieve with this PR?

We want to fix the bug by implementing the necessary changes in the DTR Search Logic
 
# What is new?

## Added
- Fixed the timeout time for each negotiation
- Added check for empty or null contractIds with retrial
- Added descriptive logs to search and create methods

## Issues Fixed
- Fixed the long waiting time by implementing timeout when doing the negotiation
- Fixed the null contract ids creation

### CODE REVIEW

File: charts/digital-product-pass/values-beta.yaml [hunk 1]
```diff
@@ -78,12 +78,11 @@ backend:
...
-    helm:
-      clientId: <path:material-pass/data/beta/backend#clientId>
-      clientSecret: <path:material-pass/data/beta/backend#clientSecret>
-      xApiKey: <path:material-pass/data/beta/edc/oauth#api.key>
-      participantId:  <path:material-pass/data/beta/edc/participant#bpnNumber>
+  edc:
+    clientId: <path:material-pass/data/dev/backend#clientId>
+    clientSecret: <path:material-pass/data/dev/backend#clientSecret>
+    xApiKey: <path:material-pass/data/dev/edc/oauth#api.key>
+    participantId:  <path:material-pass/data/dev/edc/participant#bpnNumber
```
Comment (saudkhan116) [inline charts/digital-product-pass/values-beta.yaml:85]: missing closing parenthesis ">" for the participantId.

File: charts/digital-product-pass/values-dev.yaml [hunk 2]
```diff
@@ -78,12 +78,11 @@ backend:
...
-    helm:
-      clientId: <path:material-pass/data/dev/backend#clientId>
-      clientSecret: <path:material-pass/data/dev/backend#clientSecret>
-      xApiKey: <path:material-pass/data/dev/edc/oauth#api.key>
-      participantId:  <path:material-pass/data/dev/edc/participant#bpnNumber>
+  edc:
+    clientId: <path:material-pass/data/dev/backend#clientId>
+    clientSecret: <path:material-pass/data/dev/backend#clientSecret>
+    xApiKey: <path:material-pass/data/dev/edc/oauth#api.key>
+    participantId:  <path:material-pass/data/dev/edc/participant#bpnNumber
```
Comment (saudkhan116) [inline charts/digital-product-pass/values-dev.yaml:85]: same as above

**Review by saudkhan116 — CHANGES_REQUESTED** [review]
Everything is fine, except a small missing/typo

File: charts/digital-product-pass/values-beta.yaml (hunk 1 above)
Comment (matbmoser) [reply charts/digital-product-pass/values-beta.yaml:85]: values beta has dev secret references

File: charts/digital-product-pass/values-dev.yaml (hunk 2 above)
Comment (matbmoser) [reply charts/digital-product-pass/values-dev.yaml:85]: This is also correct is missing a >

**Comment by matbmoser** [comment]
Here we need to add the change logs

File: charts/digital-product-pass/values-dev.yaml (hunk 2 above)
Comment (saudkhan116) [reply charts/digital-product-pass/values-dev.yaml]: The typo is fixed, closing this conversation

File: charts/digital-product-pass/values-beta.yaml (hunk 1 above)
Comment (saudkhan116) [reply charts/digital-product-pass/values-beta.yaml]: The typo is fixed, closing this conversation

**Review by saudkhan116 — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -24,6 +24,28 @@
 
 The changelog format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [unreleased]
```
Comment (matbmoser) [inline CHANGELOG.md:27]: changed to released and it fine :)

Comment (saudkhan116) [reply CHANGELOG.md]: done 💯

**Comment by saudkhan116** [comment]
The ip chceks are executed and nothing was found

**Pull request merged**